New chapter in Applications: Facial Expression Recognition

incud opened the pull request:
# Changes made

This PR adds one chapter to the Qiskit Textbook, under section 4.2 of the textbook (might possibly be 4.2.3).

# Justification

This chapter enriches the showcase of possible applications of quantum software to image processing, following [Quantum Image Processing - FRQI and NEQR Image Representations](https://qiskit.org/textbook/ch-applications/image-processing-frqi-neqr.html) and possibly [Quantum Edge Detection](https://github.com/qiskit-community/qiskit-textbook/pull/1084). 

The chapter shows the implementation of a simple classifier applied to the Facial Expression Recognition problem. Some techniques for preprocessing the faces are shown. The work is adapted from [here](https://link.springer.com/article/10.1007/s42484-020-00035-5). 

The content of this chapter was briefly, informally discussed with @grossiM and @robertloredo. Dear reviewers, any suggestion is always welcome. I hope to hear from you and together improve the chapter to its best.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/qiskit-community/qiskit-textbook?pullRequest=1179) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/qiskit-community/qiskit-textbook?pullRequest=1179) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/qiskit-community/qiskit-textbook?pullRequest=1179) it.</sub>

divshacker commented on content/ch-applications/facial-expresson-recognition.ipynb:
```diff
@@ -0,0 +1,808 @@
+   "outputs": [
+    {
+     "data": {
+      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAaMAAAGQCAYAAAADew/3AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/Z1A+gAAAACXBIWXMAAAxOAAAMTgF/d4wjAAEAAElEQVR4nOz917JlyZWuiX3D3adaauuIzEgBoIBCie4+LY6RNNL4FjSSN7Sm8Yr3fck34COQV/00vGE3jaQ1zc6pUwUURKoQWy01hQteDPe51k6gBNBtlpZ2woHI2BGx1pw+fboP8Y9/jCEpJT6Oj+Pj+Dg+jo/jhxzmh57Ax/FxfBwfx8fxcXxURh/Hx/FxfBwfxw8+Piqjj+Pj+Dg ...
```
Hey I think this latex code is not rendering properly.

incud replied:
@divshacker Thank you for your comment! I'm checking it right now.

incud replied:
@divshacker This is strange because with Jupyter-Lab (2.3.1 on Windows 10, with Python 3.8.5) I can visualize everything correctly. I have re-wrote the formulas in one line (previously most formulas were spread on multiple lines). I still can visualize everything correctly. Can you tell me if it's ok now? Thank you very much

divshacker replied:
Yes it is ok now

divshacker reviewed (commented):
![Screenshot from 2021-06-29 14-50-59](https://user-images.githubusercontent.com/55018955/123772540-8b3de100-d8e9-11eb-874e-30535061cfa8.png)
This is the screenshot of your uploaded ipynb file where the latex code is not rendered properly

divshacker commented on content/ch-applications/facial-expresson-recognition.ipynb:
```diff
@@ -0,0 +1,794 @@
+   "outputs": [
+    {
+     "data": {
+      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAaMAAAGQCAYAAAADew/3AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/Z1A+gAAAACXBIWXMAAAxOAAAMTgF/d4wjAAEAAElEQVR4nOz917JlyZWuiX3D3adaauuIzEgBoIBCie4+LY6RNNL4FjSSN7Sm8Yr3fck34COQV/00vGE3jaQ1zc6pUwUURKoQWy01hQteDPe51k6gBNBtlpZ2woHI2BGx1pw+fboP8Y9/jCEpJT6Oj+Pj+Dg+jo/jhxzmh57Ax/FxfBwfx8fxcXxURh/Hx/FxfBwfxw8+Piqjj+Pj+Dg ...
```
Hey I think this one is left for fixing @incud

divshacker replied:
I think this is left for fixing @incud

divshacker replied:
I think This is left for fixing @incud

divshacker replied:
I think This is left for fixing @incud

divshacker commented:
![Screenshot from 2021-07-01 20-40-32](https://user-images.githubusercontent.com/55018955/124147842-b0c71800-daac-11eb-91a5-05c7e2940640.png)
I think This is left for fixing @incud

incud commented:
> ![Screenshot from 2021-07-01 20-40-32](https://user-images.githubusercontent.com/55018955/124147842-b0c71800-daac-11eb-91a5-05c7e2940640.png)
> I think This is left for fixing @incud
> All other previous fixes looks good

Fixed. Thank you :)

frankharkins commented:
@incud this written clearly and I really appreciate the docstrings in the code. Can you include some comparison of how this performs compared to the best-known classical algorithm in the introduction please?

incud commented:
@frankharkins thank you for your feedback! Of course, I'm fixing it right now :)

incud commented:
Thank you for your patience! I have updated the _data/toc.yaml and added the comparison between classical and quantum algorithm as asked by @frankharkins :)

incud commented:
I have updated the notebook with the feedback given by @frankharkins :)

frankharkins reviewed (changes requested):
I think this is a great chapter and really well written. The code especially is clear and I appreciate you've included docstrings. Two final small things before merging are:
1. > Finally, we need a procedure that encodes our data. To this purpose we use the Initialize gate. Note that Initialize...
   
   From what I understand, `Initialize` can be very inefficient and the speed of the algorithm is dependent on it being efficient. If so, we need to be open with readers that this is a bottleneck and an open problem. Please correct me if I'm wrong.
2. Can you do a quick search & replace `ancilla / ancillary` -> `auxiliary` please?

incud commented:
Thank you @frankharkins for your feedback. I've just implemented the requested changes :)

frankharkins reviewed (approved):
Thank you so much!
